site/static/examples/kind-with-registry.sh

```diff
@@ -10,30 +10,46 @@ if [ "$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true
     registry:2
 fi
 
-# create a cluster with the local registry enabled in containerd
+# 2. Create kind cluster with containerd registry config dir enabled
+# TODO: kind will eventually enable this by default and this patch will
+# be unnecessary.
+#
+# See:
+# https://github.com/kubernetes-sigs/kind/issues/2875
+# https://github.com/containerd/containerd/blob/main/docs/cri/config.md#registry-configuration
+# See: https://github.com/containerd/containerd/blob/main/docs/hosts.md
 cat <<EOF | kind create cluster --config=-
 kind: Cluster
 apiVersion: kind.x-k8s.io/v1alpha4
 containerdConfigPatches:
 - |-
   [plugins."io.containerd.grpc.v1.cri".registry]
-    # ensure config_path is disabled so the config below is respected
-    # TODO: kind will eventually migrate to using config_path
-    # see:
-    # https://github.com/kubernetes-sigs/kind/issues/2875
-    # https://github.com/containerd/containerd/blob/main/docs/cri/config.md#registry-configuration
-    config_path = ""
+    config_path = "/etc/containerd/certs.d"
+EOF
 
-  [plugins."io.containerd.grpc.v1.cri".registry.mirrors."localhost:${reg_port}"]
-    endpoint = ["http://${reg_name}:5000"]
+# 3. Add the registry config to the nodes
+#
+# This is necessary because localhost resolves to loopback addresses that are
+# network-namespace local.
+# In other words: localhost in the container is not localhost on the host.
+#
+# We want a consistent name that works from both ends, so we tell containerd to
+# alias localhost:${reg_port} to the registry container when pulling images
+REGISTRY_DIR="/etc/containerd/certs.d/localhost:${reg_port}"
+for node in $(kind get nodes); do
+  docker exec "${node}" mkdir -p "${REGISTRY_DIR}"
+  cat <<EOF | docker exec -i "${node}" cp /dev/stdin "${REGISTRY_DIR}/hosts.toml"
```

Comment:
there might be some clever way to do this with a single command, but this is how we write files internally to kind currently.

Reply:
`docker cp` ?

Reply:
Will cp create missing directory?

Reply:
nevermind, I misunderstood the comment, we need to replace the var and create a folder

Reply:
Yeah, we'd still have to mkdir and I don't think you can stain pipe a regular file in to docker cp, it takes a tar